PhysicsTools/PatAlgos/python/slimming/customizeMiniAOD_HcalFixLegacy2016.py

```diff
@@ -0,0 +1,134 @@
+from PhysicsTools.PatAlgos.tools.helpers import getPatAlgosToolsTask
+
+def loadJetMETBTag(process):
+
+    task = getPatAlgosToolsTask(process)
+
+    import RecoJets.Configuration.RecoPFJets_cff
+    process.ak4PFJetsCHS = RecoJets.Configuration.RecoPFJets_cff.ak4PFJetsCHS.clone()
+    task.add(process.ak4PFJetsCHS)
+    # need also the non-CHS ones as they are used to seed taus
+    process.ak4PFJets = RecoJets.Configuration.RecoPFJets_cff.ak4PFJets.clone()
+    task.add(process.ak4PFJets)
+    process.ak8PFJetsCHS = RecoJets.Configuration.RecoPFJets_cff.ak8PFJetsCHS.clone()
+    task.add(process.ak8PFJetsCHS)
+    process.load("RecoMET.METProducers.PFMET_cfi")
+    task.add(process.pfMet)
+    process.load("RecoJets.JetAssociationProducers.ak4JTA_cff")
+    task.add(process.ak4JetTracksAssociatorAtVertexPF)
+    process.load("RecoBTag.ImpactParameter.impactParameter_cff")
+    task.add(process.impactParameterTask)
+    process.load("RecoBTag.SecondaryVertex.secondaryVertex_cff")
+    task.add(process.secondaryVertexTask)
+    process.load("RecoBTag.SoftLepton.softLepton_cff")
+    task.add(process.softLeptonTask)
+    process.load("RecoBTag.Combined.combinedMVA_cff")
+    task.add(process.combinedMVATask)
+    process.load("RecoBTag.CTagging.cTagging_cff")
+    task.add(process.cTaggingTask)
+    process.load("RecoVertex.AdaptiveVertexFinder.inclusiveVertexing_cff")
+    task.add(process.inclusiveVertexingTask)
+    task.add(process.inclusiveCandidateVertexingTask)
+    task.add(process.inclusiveCandidateVertexingCvsLTask)
+
+
+def cleanPfCandidates(process, verbose=False):
+    task = getPatAlgosToolsTask(process)
+
+    #add producer at the beginning of the schedule
+    process.load("CommonTools.ParticleFlow.pfCandidatesBadHadRecalibrated_cfi")
+    task.add(process.pfCandidatesBadHadRecalibrated)
+
+    replacePFCandidates = MassSearchReplaceAnyInputTagVisitor("particleFlow", "pfCandidatesBadHadRecalibrated", verbose=verbose)
+    for everywhere in [ process.producers, process.filters, process.analyzers, process.psets, process.vpsets ]:
+        for name,obj in everywhere.iteritems():
+            if obj != process.pfCandidatesBadHadRecalibrated:
+                replacePFCandidates.doIt(obj, name)
+
+
+    process.load("CommonTools.ParticleFlow.pfEGammaToCandidateRemapper_cfi")
+    task.add(process.pfEGammaToCandidateRemapper)
+    process.pfEGammaToCandidateRemapper.pf2pf = cms.InputTag("pfCandidatesBadHadRecalibrated")
+    process.reducedEgamma.gsfElectronsPFValMap = cms.InputTag("pfEGammaToCandidateRemapper","electrons")
+    process.reducedEgamma.photonsPFValMap      = cms.InputTag("pfEGammaToCandidateRemapper","photons")
+
+
+    #add bugged conditions to GT for comparison
+    process.GlobalTag.toGet.append(cms.PSet(
+        record = cms.string("HcalRespCorrsRcd"),
+        label = cms.untracked.string("bugged"),
+        tag = cms.string("HcalRespCorrs_v6.3_offline") #to be replaced with proper bugged tag
```

Comment:
this should be in a GT.
@lpernie @arunhep 
please clarify if this was already requested

Reply:
that's the tag with the bugged conditions which are compared with the fixed respcorrs available in the GT so it is supposed to stay as it is.
I've asked an update the the 94x GT at https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/3495/1/1/1/1/1.html

Reply:
@lpernie @arunhep
please clarify if/when this payload will be in the GT

Reply:
I've asked the Hcal DB guys to queue the `HcalRespCorrs_v6.3_offline` tag with label `bugged` to `94X_dataRun2_Queue` and `94X_dataRun2_relval_Queue`